astropy/cosmology/_src/utils.py

```diff
@@ -66,7 +67,11 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
-def aszarr(z):
+class HasShape(Protocol):
+    shape: tuple[int, ...]
+
+
+def aszarr(z: Number | np.generic | HasShape | ArrayLike) -> NDArray[Any]:
```

Comment:
This signature is clearly not right.

Reply:
Yes, it should include `Number | np.generic | `. I've updated it.
But most accurately it should return a DuckArray. We only support NDArray at the moment, so I think that's fine for now.